### PR TITLE
Update deployment for OIDC

### PR DESCRIPTION
**Pull request by IAlibay**
Towards: https://github.com/MDAnalysis/mdanalysis/issues/4326

PR Checklist
------------
 - [ ] CHANGELOG updated?

### CODE REVIEW

**Comment by IAlibay** [comment]
Trusted publishers have been set.

**Comment by IAlibay** [comment]
🤦🏽 why is py3.12 being called here - is compatible python not work?

**Comment by IAlibay** [comment]
There we go, that's CI fixed - I'm not sure how I feel about this properly, but maybe this is the best approach.

File: .github/workflows/gh-ci.yaml [hunk 1]
```diff
@@ -35,6 +36,8 @@ jobs:
 
       - id: get-compatible-python
         uses: MDAnalysis/mdanalysis-compatible-python@main
+        with:
+          release: "latest"
```
Comment (orbeckst) [inline .github/workflows/gh-ci.yaml:40]: That's the current py 3.12 hack... https://github.com/MDAnalysis/cookiecutter-mdakit/pull/93

**Review by orbeckst — APPROVED** [review]
Thanks for updating to trusted publishers _and_ the 3.12 hack.

**Pull request merged**